MapboxCoreNavigation/Route.swift

```diff
@@ -2,7 +2,7 @@ import MapboxDirections
 
 extension Route {
     
-    var json: String? {
+    public var json: String? {
```

Comment:
Per https://github.com/mapbox/mapbox-navigation-ios/pull/2275#discussion_r358536495, this is potentially an expensive operation, so we should communicate that to clients by exposing it as a method instead of a computed variable. It’s also very specific to the needs of MapboxNavigationNative – returning a string instead of Data or a JSON-formatted dictionary – so I’m kind of thinking it would be better for any application that needs this functionality to reimplement it locally. JSONEncoder is public API, part of the Swift standard library.